service/src/test/java/bio/terra/workspace/pact/LzApiTest.java

```diff
@@ -0,0 +1,612 @@
+package bio.terra.workspace.pact;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.anEmptyMap;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.DslPart;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.PactSpecVersion;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.landingzone.library.landingzones.deployment.ResourcePurpose;
+import bio.terra.workspace.amalgam.landingzone.azure.HttpLandingZoneService;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneServiceAuthorizationException;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneServiceNotFoundException;
+import bio.terra.workspace.app.configuration.external.LandingZoneServiceConfiguration;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneParameter;
+import bio.terra.workspace.generated.model.ApiAzureLandingZoneResourcesPurposeGroup;
+import bio.terra.workspace.generated.model.ApiJobReport;
+import io.opentelemetry.api.OpenTelemetry;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.http.HttpStatus;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+public class LzApiTest {
+
+  HttpLandingZoneService landingZoneService;
+
+  static final Map<String, String> contentTypeJsonHeader =
+      Map.of("Content-Type", "application/json");
+  static final UUID landingZoneId = UUID.fromString("a051c5f8-9de8-4b4f-a453-8f9645df9c7b");
+  static final UUID asyncJobId = UUID.fromString("3b0db775-1916-42d2-aaeb-ebc7aea69eca");
+  static final UUID billingProfileId = UUID.fromString("4955757a-b027-4b6c-b77b-4cba899864cd");
+  static final UUID azureResourceId = UUID.fromString("e3327645-1e8f-454d-bdae-fe06b4762542");
+
+  static DslPart jobReportShape =
+      new PactDslJsonBody()
+          .stringType("id")
+          .stringType("description")
+          .stringMatcher("status", "RUNNING|SUCCEEDED|FAILED")
```

Comment:
Normally, I would say they should be looser, and only declare what the service actually uses, but since WSM is basically acting as a proxy for LZS, defining all the fields makes sense.

If any fields are optional though, we should write separate validations for when they are present, and omit them otherwise.

For example, `completed` won't be present in jobs that are still running.
So for the default case, we should omit it.
We still want to validate the correct type of `completed`, so for that we should write a test that checks for it in a case where it's required to be present.

Reply:
Good point; I've taken a whack at that [here](https://github.com/DataBiosphere/terra-workspace-manager/pull/1714/files#diff-ed7c73cdb1082bca22090bc2b30b66a80fe68e3f9f02179ff4e2b10975ffa5c9R45-R76), let me know what you think.